src/api/RequestApi.ts

```diff
@@ -0,0 +1,198 @@
+import { spWebContext } from "../providers/SPWebContext";
+import { ApiError } from "./InternalErrors";
+import { IItemUpdateResult } from "@pnp/sp/items";
+import { emptype } from "../constants/EmpTypes";
+import { worklocation } from "../constants/WorkLocations";
+import { SPPersona } from "../components/PeoplePicker/PeoplePicker";
+
+// create PnP JS response interface for the InForm
+// This extends the IInForm -- currently identical, but may need to vary when pulling in SPData
+type IResponseItem = IInForm;
+
+// create IItem item to work with it internally
+export type IInForm = {
+  /** Required - Will be -1 for NewForms that haven't been saved yet */
+  Id: number;
+  /** Required - Contains the Employee's Name */
+  empName: string;
+  /** Required - Employee's Type valid values are:
+   * 'civ' - for Civilian Employees
+   * 'mil' - for Military Employees
+   * 'ctr' - for Contracted Employees
+   */
+  empType: emptype;
+  /** Required - The Employee's Grade/Rank.  Not applicable if 'ctr' */
+  gradeRank: string;
+  /** Required - Possible values are 'local' and 'remote'  */
+  workLocation: worklocation;
+  /** Required - The Employee's Office */
+  office: string;
+  /** Required - Can only be 'true' if it is a New to USAF Civilain.  Must be 'false' if it is a 'mil' or 'ctr' */
+  // TODO - Look into making this a Type or Leveraging SharePoint Type -- Think they possibly use yes/no instead of true/false
+  isNewCiv: "yes" | "no";
+  /** Required - The user's previous organization.  Will be "" if isNewCiv is false */
+  prevOrg: string;
+  /** Required - The user's Estimated Arrival Date */
+  eta: Date;
+  /** Required - The Superviosr/Gov Lead of the employee */
+  supGovLead: SPPersona[];
+};
+
+/** For new forms, allow certain fields to be blank or undefined to support controlled components */
+export type INewInForm = Omit<
+  IInForm,
+  "empType" | "workLocation" | "eta" | "supGovLead" | "isNewCiv"
+> & {
+  empType: emptype | "";
+  workLocation: worklocation | "";
+  eta: Date | undefined;
+  supGovLead: SPPersona[] | undefined;
+  isNewCiv: "yes" | "no" | "";
```

Comment:
Should these be empty strings or nulls?
isNewCiv I could see being an empty string assuming you don't change it to a boolean

Reply:
I think with controlled components, we want the values to be "" not null
https://reactjs.org/docs/forms.html#controlled-input-null-value